### PR TITLE
[Hub Generated] Review request for Microsoft.BareMetalInfrastructure to add version preview/2023-08-04-preview

### PR DESCRIPTION
**Pull request by alexandraavendano**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://openapihub.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/alexandraavendano-baremetalinfrastructure-Microsoft.BareMetalInfrastructure-2023-08-04-preview...main/).
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I have reviewed the general guidance on the spec PR review process: https://aka.ms/specprreview.
- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data-plane related specifications.
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy).
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can request review from an ARM API Review board.

### ARM API changes review

- If you want for the ARM team to review this PR, you must add the `ARMReview` label. 
- The automation may automatically add the `ARMReview` label, if appropriate.  
  If this happens, proceed according to guidance given in GitHub comments also added by the automation.

### Breaking change review

If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), 
follow the process outlined in the [High-level Breaking Change Process doc](https://aka.ms/breakingchangesprocess#high-level-breaking-change-process).
      
## Getting help

- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For additional help, see https://aka.ms/azsdk/support/spectools.

### CODE REVIEW

**Review by pabowers — APPROVED** [review]
LGTM

**Comment by ms-zhenhua** [comment]
Please reach out to Azure Breaking Changes Reviewers azbreakchangereview@microsoft.com to get a breaking changes approval and ARM can review the changes afterwards. Once you have the approval, please remove the `ARMChangesRequested` label from the PR to make the PR visible to the ARM reviewers.

File: specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json
Comment (alexandraavendano) [reply specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json]: Thanks for pointing that out. I just push the expected "restarting" message.

File: specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json
Comment (8Gitbrix) [inline specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json:926]: I'm assuming I should also add lun back to the ALI spec?

File: specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json
Comment (ms-henglu) [inline specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json:76]: please always use the latest, current is "v5". please also update other places.

File: specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json
Comment (ms-henglu) [inline specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json:751]: If using the latest TrackedResource definition, there's no need to define the systemData here.

File: specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json
Comment (ms-henglu) [inline specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json:782]: Is it an azure resource ID? If so, please add "format": "arm-id" and "x-ms-arm-id-details".

https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details

File: specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json
Comment (ms-henglu) [inline specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json:1142]: ARM recommends string enum over boolean for many reasons. Consider something like activation: { Enabled, Disabled } (or similar with better names). Ref: https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review.

File: specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json
Comment (ms-henglu) [inline specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json:1156]: Consider model it as an enums if the property has a finite set of values?
refs: https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi015-avoid-modelling-a-property-as-string-type-when-it-should-be-an-enums

File: specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json
Comment (alexandraavendano) [reply specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json:926]: Please do not add the lun property in ALI. This object isn't present in our domain model. I've retained it because its removal would result in a breaking change. However, in ALI, removing the 'lun' doesn't constitute a breaking change. We can address the 'lun' issue in a subsequent iteration for BMI.

File: specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json
Comment (alexandraavendano) [reply specification/baremetalinfrastructure/resource-manager/Microsoft.BareMetalInfrastructure/preview/2023-08-04-preview/baremetalinfrastructure.json:782]: azureBareMetalInstanceId property is a unique uuid. We don't use it in any API as parameter. It's just a unique identifier that allow us to check uniqueness between assigned server and customer.

**Comment by kazrael2119** [comment]
this js sdk breaking is caused by this pr

**Review by raosuhas — APPROVED** [review]

**Pull request merged**
